src/ProjectService.Business/Commands/Task/GetTaskCommand.cs

```diff
@@ -192,46 +194,57 @@ private List<Guid> GetUsersIds(DbTask task, Guid? parentTaskAssignedTo)
       _logger = logger;
       _rcGetCompanyEmployee = rcGetCompanyEmployee;
       _usersDataRequestClient = userRequestClient;
-      _rcImages = rcImages;
-      _imageMapper = imageMapper;
+      _cache = cache;
     }
 
-    public OperationResultResponse<TaskResponse> Execute(Guid taskId, bool isFullModel = true)
+    public async Task<OperationResultResponse<TaskResponse>> Execute(Guid taskId, bool isFullModel = true)
     {
-      OperationResultResponse<TaskResponse> response = new OperationResultResponse<TaskResponse>();
+      var errors = new List<string>();
+
       DbTask task = _taskRepository.Get(taskId, isFullModel);
 
-      if (task == null)
-      {
-        _httpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+      (bool hasRights, DepartmentData department) = await Authorization(task.ProjectId, errors);
 
-        response.Status = OperationResultStatusType.Failed;
-        response.Errors.Add($"Task with Id {taskId} was not found.");
+      if (!hasRights)
+      {
+        _httpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
 
-        return response;
+        return new OperationResultResponse<TaskResponse>
+        {
+          Status = OperationResultStatusType.Failed,
+          Errors = new() { "Not enough rights." }
```

Comment:
errors already instantiated -> use add. i think it faster then create new object

Reply:
fixed